Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -789,16 +662,16 @@ void ReflectometryReductionOne2::findDetectorGroups() {
   }
 
   for (const auto &group : m_detectorGroups) {
-    for (const auto &spIdx : group) {
-      if (m_spectrumInfo->isMonitor(spIdx)) {
-        throw std::invalid_argument("A detector is expected at spectrum " +
-                                    std::to_string(spIdx) +
-                                    ", found a monitor");
+    for (const auto &wsIdx : group) {
+      if (m_spectrumInfo->isMonitor(wsIdx)) {
+        throw std::invalid_argument(
+            m_processingInstructions +
```

Comment:
The new error message is misleading because it's reporting that it's an invalid spectrum number when actually it may be a valid spectrum number, it's just not a valid detector because it's a monitor. Also we should report the specific spectrum number / workspace index that's the problem, rather than the whole processing instructions string.

I think the original error message was better here, it just needed to be changed to report as `A detector is expected at workspace index...` rather than `A detector is expected at spectrum...`. Or alternatively (and probably nicer) we could report the spectrum number but we would have to convert `wsIdx` to a spectrum number.